src/focused-stack/index.ts

```diff
@@ -16,10 +16,16 @@ interface HandlerGroup {
 export class FocusedStack {
   private keyGenId: number;
   private stack: HandlerGroup[];
+  private tearDownHandler: (() => void) | null;
+  private timeOut: number;
+  private clock: number | null; // the returned timerID value from setTimeout() is a positive integer
 
-  constructor() {
+  constructor(timeOut: number) {
```

Comment:
To allow for future expansion, I think I'd prefer an object here like:

```ts
export interface FocusedStackOptions {
  timeout: number;
}
```

And then have the constructor take a partial option object with defaults like:

```ts
constructor(options: Partial<FocusedStackOption> = {}) {
  const { timeout = 3000 } = options;
  this.timeout = timeout;
}
```